tests/jerry/es2015/regression-test-issue-3589.js

```diff
@@ -0,0 +1,27 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+try {
+  [this,000000000,this,99999999=9999999];
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
```

Comment:
Is it ok to expect a ReferenceError? It should be a SyntaxError (as assigning a value to a constant is not valid)

Reply:
That's also an es5.1 bug, so the tc should be moved as well. However ES6 static semantic is not implemented  yet, therefore I'd suggest to keep the ES5.1 solution for now.